### ISSUE TITLE
Error when updating to Craft 3.1.3

### ISSUE THREAD

**Issue by rensverschuren** [engram-design/SuperTable#229]
### Description

When I update Craft and want to run the migrations, I get the following error:

```
Database Exception: SQLSTATE[42P07]: Duplicate table: 7 ERROR: relation "stc_3_checklist" already exists
The SQL being executed was: ALTER TABLE "stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist" RENAME TO "stc_3_checklist"

Migration: verbb\supertable\migrations\m190120_000000_fix_supertablecontent_tables

Output:

> Super Table field #6 has content table ...
> Trying to create missing content table for #6 ...
> Generated table name {{%stc_checklist}} ...
> update in {{%fields}} ... done (time: 0.039s)
> Local field table name {{%stc_checklist}} ...
> Updated Super Table field after content table creation ...

> Super Table field #9 has content table ...
> Trying to create missing content table for #9 ...
> Generated table name {{%stc_usps}} ...
> update in {{%fields}} ... done (time: 0.125s)
> Local field table name {{%stc_usps}} ...
> Updated Super Table field after content table creation ...

> Super Table field #16 has content table ...
> Trying to create missing content table for #16 ...
> Generated table name {{%stc_seolinks}} ...
> update in {{%fields}} ... done (time: 0.027s)
> Local field table name {{%stc_seolinks}} ...
> Updated Super Table field after content table creation ...

> Super Table field #36 has content table ...
> Trying to create missing content table for #36 ...
> Generated table name {{%stc_3_checklist}} ...
> update in {{%fields}} ... done (time: 0.035s)
> Local field table name {{%stc_3_checklist}} ...
> Updated Super Table field after content table creation ...

> Super Table field #28 has content table ...
> Trying to create missing content table for #28 ...
> Generated table name {{%stc_2_checklist}} ...
> update in {{%fields}} ... done (time: 0.060s)
> Local field table name {{%stc_2_checklist}} ...
> Updated Super Table field after content table creation ...

> Incorrect nested content table found {{%stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist}} ...
> drop foreign key stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist_elementId_fk from table stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist ... done (time: 0.005s)
> drop foreign key stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist_siteId_fk from table stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist ... done (time: 0.007s)
> drop index st_34689587-43a0-40b6-b685-07ef5c_checkli_element_siteI_un_id on {{%stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist}} ... done (time: 0.003s)
> rename table stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist to stc_3_checklist ...Exception: SQLSTATE[42P07]: Duplicate table: 7 ERROR: relation "stc_3_checklist" already exists
The SQL being executed was: ALTER TABLE "stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist" RENAME TO "stc_3_checklist" (/app/vendor/yiisoft/yii2/db/Schema.php:664)
#0 /app/vendor/yiisoft/yii2/db/Command.php(1263): yii\db\Schema->convertException(Object(PDOException), 'ALTER TABLE "st...')
#1 /app/vendor/yiisoft/yii2/db/Command.php(1075): yii\db\Command->internalExecute('ALTER TABLE "st...')
#2 /app/vendor/yiisoft/yii2/db/Migration.php(340): yii\db\Command->execute()
#3 /app/vendor/craftcms/cms/src/helpers/MigrationHelper.php(171): yii\db\Migration->renameTable('stc_34689587-43...', 'stc_3_checklist')
#4 /app/vendor/verbb/super-table/src/migrations/m190120_000000_fix_supertablecontent_tables.php(137): craft\helpers\MigrationHelper::renameTable('{{%stc_34689587...', '{{%stc_3_checkl...', Object(verbb\supertable\migrations\m190120_000000_fix_supertablecontent_tables))
#5 /app/vendor/craftcms/cms/src/db/Migration.php(56): verbb\supertable\migrations\m190120_000000_fix_supertablecontent_tables->safeUp()
#6 /app/vendor/craftcms/cms/src/db/MigrationManager.php(243): craft\db\Migration->up(true)
#7 /app/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(verbb\supertable\migrations\m190120_000000_fix_supertablecontent_tables))
#8 /app/vendor/craftcms/cms/src/services/Updates.php(223): craft\db\MigrationManager->up()
#9 /app/vendor/craftcms/cms/src/controllers/BaseUpdaterController.php(445): craft\services\Updates->runMigrations(Array)
#10 /app/vendor/craftcms/cms/src/controllers/UpdaterController.php(208): craft\controllers\BaseUpdaterController->runMigrations(Array, 'restore-db')
#11 [internal function]: craft\controllers\UpdaterController->actionMigrate()
#12 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /app/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#14 /app/vendor/craftcms/cms/src/web/Controller.php(109): yii\base\Controller->runAction('migrate', Array)
#15 /app/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('migrate', Array)
#16 /app/vendor/craftcms/cms/src/web/Application.php(297): yii\base\Module->runAction('updater/migrate', Array)
#17 /app/vendor/craftcms/cms/src/web/Application.php(683): craft\web\Application->runAction('updater/migrate')
#18 /app/vendor/craftcms/cms/src/web/Application.php(243): craft\web\Application->_processUpdateLogic(Object(craft\web\Request))
#19 /app/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#20 /app/web/index.php(21): yii\base\Application->run()
#21 {main}
```

### Steps to reproduce

1. `Composer update`
2. Run the migrations

### Additional info

- Plugin version: 2.1.5.1
- Craft version: 3.1.3

**Comment by rensverschuren** [comment]
There seem to be two empty tables:

1. `stc_34689587-43a0-40b6-b685-07ef5c50d30d_checklist`
2. `stc_8df1df3d-bd46-4bd0-90a2-10c371610383_checklist`

When I delete these tables, the migrations run fine. No data seems to be lost.

**Comment by engram-design** [comment]
Yep, those tables can be safely deleted without data loss. They were created through the incorrect migration introduced in 3.1.x. This migration is trying to rename the incorrect table back to `stc_3_checklist `. This error should be fixed in [2.1.5.2](https://github.com/verbb/super-table/releases/tag/2.1.5.2) as well.

If you run into any issues after, or just want to check up on things, be sure to run the new `http://mysite.com/actions/super-table/plugin/check-content-tables` action endpoint to check Super Table content tables for any issues.

**Comment by rensverschuren** [comment]
Cool, thank you!
